hdf/src/hdf.h

```diff
@@ -138,6 +138,20 @@ typedef intn (*hdf_termfunc_t)(void); /* termination function typedef */
 
 /* .................................................................. */
 
+/* Publicly useful macros for tests/examples.  */
+
+/* Used to make certain a checked value _is_not_ a specific value.
+ * If the checked value matches the specific value, print an error message. */
+#define CHECK_NOT_VAL(ret, val, where)                                                                       \
+    do {                                                                                                     \
+        if (ret == val) {                                                                                    \
+            printf("*** ERROR from %s is %ld at line %4d in %s\n", where, (long)ret, (int)__LINE__,          \
+                   __FILE__);                                                                                \
+        }                                                                                                    \
+    } while (0)
+
+/* .................................................................. */
```

Comment:
I don't think we should put this in the public API. I'm normally against duplicating code, but in this case I'd prefer duplicating it in each set of examples.

Reply:
In particular, this is going to be visible in the test's tutil.h file, which seems like it has a high probability of causing problems if we were to refactor things.